Automattic/wp-calypso
# Purchases: Use new `is_locked` property for hiding management actions

Fixes https://github.com/Automattic/wp-calypso/issues/56498

#### Changes proposed in this Pull Request

As discussed in p3hLNG-18B-p2, we cannot use the blog stickers endpoint to determine whether a purchase is locked. Instead, in D67879-code we're exposing that info to all endpoints returning a purchase object. This PR replaces all the previous stickers logic with an approach that checks the new `is_locked` property.

#### Testing instructions

- Apply D67879-code to your WP.com sandbox.
- Sandbox the API.
- Use the Calypso live link below and append `?flags=atomic/automated-revert` at the end of the URL.
- Switch to an Atomic site.
- Go to Upgrades > Purchases.
- Remove the site plan.
- Make sure you get a success notification indicating the removal/cancellation has been queued.
- Make sure you cannot remove/cancel the plan again (none of the management buttons such as "Cancel", "Renew", etc shouldn't show up).
- Make sure you can remove/cancel other purchases.
- Manually run the `atomic_subscription_revert_after_lossless_import` async job (see testing instructions in D66763-code).
- Reload Calypso.
- Make sure the Atomic site is on a Free plan now.
